consonance-webservice/src/main/java/io/consonance/webservice/ConsonanceWebserviceApplication.java

```diff
@@ -88,24 +101,43 @@ public void run(ConsonanceWebserviceConfiguration configuration, Environment env
         environment.healthChecks().register("template", healthCheck);
 
         final JobDAO dao = new JobDAO(hibernate.getSessionFactory());
-        final HttpClient httpClient = new HttpClientBuilder(environment).using(configuration.getHttpClientConfiguration()).build(getName());
+        final ProvisionDAO provisionDAO = new ProvisionDAO(hibernate.getSessionFactory());
+        final ConsonanceUserDAO userDAO = new ConsonanceUserDAO(hibernate.getSessionFactory());
 
-        environment.jersey().register(new JobResource(dao));
+        environment.getObjectMapper().setPropertyNamingStrategy(PropertyNamingStrategy.CAMEL_CASE_TO_LOWER_CASE_WITH_UNDERSCORES);
+        environment.getObjectMapper().disable(SerializationFeature.WRITE_DATES_AS_TIMESTAMPS);
+        environment.getObjectMapper().enable(SerializationFeature.WRITE_ENUMS_USING_TO_STRING);
+
+        environment.jersey().register(new JobResource(dao, provisionDAO, configuration.getConsonanceConfig()));
+        environment.jersey().register(new UserResource(userDAO));
 
         // swagger stuff
 
         // Swagger providers
         environment.jersey().register(ApiListingResource.class);
         environment.jersey().register(SwaggerSerializers.class);
+        LOG.info("This is our custom logger saying that we're about to load authenticators");
+        // setup authentication
+        SimpleJPAAuthenticator authenticator = new SimpleJPAAuthenticator(userDAO);
+        CachingAuthenticator<String, ConsonanceUser> cachingAuthenticator = new CachingAuthenticator<String, ConsonanceUser>(
+                environment.metrics(), authenticator, configuration.getAuthenticationCachePolicy());
+        environment.jersey().register(
+                AuthFactory.binder(new OAuthFactory<ConsonanceUser>(cachingAuthenticator, "SUPER SECRET STUFF", ConsonanceUser.class)));
 
         // optional CORS support
         // Enable CORS headers
         final FilterRegistration.Dynamic cors = environment.servlets().addFilter("CORS", CrossOriginFilter.class);
 
         // Configure CORS parameters
-        cors.setInitParameter("allowedOrigins", "*");
-        cors.setInitParameter("allowedHeaders", "X-Requested-With,Content-Type,Accept,Origin");
-        cors.setInitParameter("allowedMethods", "OPTIONS,GET,PUT,POST,DELETE,HEAD");
+        cors.setInitParameter(CrossOriginFilter.ALLOWED_ORIGINS_PARAM, "*");
+        cors.setInitParameter(CrossOriginFilter.ALLOWED_ORIGINS_PARAM, "*");
+        cors.setInitParameter(CrossOriginFilter.ALLOWED_HEADERS_PARAM,
+                "X-Requested-With,Content-Type,Accept,Content-Length,Origin,api_key,Authorization");
+        cors.setInitParameter(CrossOriginFilter.ALLOWED_METHODS_PARAM, "OPTIONS,GET,PUT,POST,DELETE,HEAD");
```

Comment:
Seems like it for swagger, they have a listing here. https://github.com/swagger-api/swagger-ui#testing-cors-support